x/payment/client/testutil/integration_test.go

```diff
@@ -130,8 +132,14 @@ func (s *IntegrationTestSuite) TestSubmitWirePayForMessage() {
 					"test: %s, output\n:", tc.name, out.String())
 
 				events := txResp.Logs[0].GetEvents()
-				for i := 0; i < len(events); i++ {
-					s.Equal("/payment.MsgPayForMessage", events[i].GetAttributes()[0].Value)
+				for _, e := range events {
+					switch e.Type {
+					case etypes.AttributeKeySpender:
+						s.Equal("payformessage", e.Type)
```

Comment:
I think this is mostly the right idea, we should be able to iterate through the different attributes, then compare with the inputs. Keep in mind that when comparing the message size, the message is padded and will always be some multiple of 256

Reply:
Since now, or eventually, we have dynamic message size (the size can be 256 or 128 or 64...). How do we determine which size is used in the block?